File: leetcode/1345-check-if-a-word-occurs-as-a-prefix-of-any-word-in-a-sentence/CheckIfAWordOccursAsAPrefixOfAnyWordInASentence.java
class Solution {
    public int isPrefixOfWord(String sentence, String searchWord) {
        int i = 0;
        int n = sentence.length();
        int w = 1;

        while (i < n) {
            int j = i;
            int m = i + searchWord.length();

            if (m > n) {
                break;
            }

            while (j < m) {
                if (sentence.charAt(j) != searchWord.charAt(j - i)) {
                    break;
                }

                j++;
            }

            if (j == m) {
                return w;
            }

            while (i < n && sentence.charAt(i) != ' ') {
                i++;
            }

            i++;
            w++;
        }

        return -1;
    }
}

public class CheckIfAWordOccursAsAPrefixOfAnyWordInASentence {
    public static void main(String[] arguments) {
        System.out.println((new Solution()).isPrefixOfWord("i love eating burger", "burg"));
    }
}

File: leetcode/1001-1999/1455-check-if-a-word-occurs-as-a-prefix-of-any-word-in-a-sentence/CheckIfAWordOccursAsAPrefixOfAnyWordInASentence.ts
function isPrefixOfWord(sentence: string, searchWord: string): number {
  let i = 0;
  let w = 1;

  while (i < sentence.length) {
    let j = i;
    let m = i + searchWord.length;

    if (m > sentence.length) {
      break;
    }

    while (j < m) {
      if (sentence.charAt(j) !== searchWord.charAt(j - i)) {
        break;
      }

      j++;
    }

    if (j === m) {
      return w;
    }

    while (j < sentence.length && sentence.charAt(j) !== " ") {
      j++;
    }

    i = j + 1;
    w++;
  }

  return -1;
};

console.log(isPrefixOfWord("this problem is an easy problem", "pro"));
console.log(isPrefixOfWord("i love eating burger", "burg"));
